### ISSUE TITLE
Reverse Transform of Segment.

### ISSUE THREAD

**Issue by ninfueng** [dmbee/seglearn#39]
Hello, I have an issue about the shape of prediction is not same to the original labels due to the segmentation process. Do you have any function to convert the shape of prediction back to the shape before the segmentation?

**Comment by dmbee** [comment]
Hi Ninnart, 

I have a function for doing this with a continuous variable. Which averages over the segments covering a sample. 

For a categorical variable, I don't think averaging is worth it. You just need to use numpy.repeat, and decide the number of repeats based on the steps variable. The beginning and end of the sequence will have a different number of repeats. 

I can put something into seglearn.util if this is what you are looking for?

**Comment by ninnartfuengfusin** [comment]
Thank for replying, please feel free to put that function into seglearn.
However, I did not sure why using np.repeat into the categorical variable case. 

I think the problem is same as you mentioned in a continuous variable, the segments is needed to reduce into a sample, but np.repeat will increase instead of reduce.

In case categorical variable, I agreed that averaging is not worth it. I think maybe the majority voting is worth it in this case?

**Comment by dmbee** [comment]
In the case of highly overlapping segments, I agree selecting highest frequency label may be a good option, but I don't have an efficient implementation at the moment and I don't know how many people if any use highly overlapping at predict time? Even still, it may only be marginally better than assigning the central samples of a segment to the predicted class, with edges cases for first and last segment.

With a continuous target (regression) it makes more sense to have a smooth output hence the averaging, but categorical variables are discontinuous anyway. 

Thoughts?

**Comment by ninnartfuengfusin** [comment]
Interest about the highly overlapping case, I never thought about this before.

I will experiment with both the highest frequency and the central (first or last) segment label, I will report the numerical results of them later. I think either one of them might be a nice feature to be included for the categorical problem.

I agree with the continuous target case, as far as I concerned, I could not see a better case than averaging.

**Comment by dmbee** [comment]
OK - I will implement averaging for continuous target. Let's see what you come up with for categorical targets and we can go from there.

**Comment by dmbee** [comment]
This was implemented for 1.2.0. 

Check out util.segmented_prediction_to_series and Pype.predict_segmented_series

Also check the plot_feature_rep.py example

Let me know if that works for you.

**Comment by ninfueng** [comment]
Sorry for not active for a while, I checked at the `plot_feature_rep.py`. From what I understand, `Pype.predict_segmented_series` is used to predict every sampling point of the input data. Quite interesting the prediction vs time graph from `plot_feature_rep.py` that you plotted. Back to the original point, do you have any methods or suggestions to convert this graph (predictions of every sampling point of data) back to the true label, `[2 6 5]` ?.

**Comment by dmbee** [comment]
I am not sure what you are asking exactly? In the example, it predicts incorrectly for exercise 2, but correctly predicts 6 and 5. Do you just want me to plot the true label as well?

**Comment by ninfueng** [comment]
From the `plot_feature_rep.py`, the length of predication from `yp0` is 5,371. However, the length of true, `y_test[1:4]` is  3. What I want is to find the way to reduce the length of `yp0` to be same as `y_test[1:4]`. In other word, I want an algorithm to convert `array([1, 1, 1, ..., 5, 5, 5])` to `[1 6 5]`.

**Comment by dmbee** [comment]
Take a look at the convenience function I added to pipe. predict_as_series
You should be able to use that, and just take the mode for a classification problem, or the median / mean for a regression problem when you want to get a single prediction for a whole series. You could also calculate any other statistics on the series as needed.

I don't think this use case has that many applications, because in general time series are not necessarily homogeneous. It makes sense to keep and analyze predictions across time either at the sampling rate of the segmentation, or at the sampling rate of the original time series.

**Comment by ninfueng** [comment]
I understand your point. In my case, I joined an competitive, and I tried to use the `seglearn` with it. The problem was the length of predictions is not equal to the length of true. Hence, I could not submit the `seglearn` predictions to the competitive. 

With `Pype.predict_segmented_series`, this makes the reducing the length of predictions to be easier. I think I can make the function to do it, and as you said there are many approaches to do it. From this, if you prefer, I will close this issue.
